app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,81 @@
+require 'twilio-ruby'
+require 'dotenv'
+
+class TweetsController < ApplicationController
+  get '/tweets' do
+    if !Helpers.is_logged_in?(session)
+      redirect to '/login'
+    else
+      @user = User.find_by_id(session[:user_id])
+      @tweets = Tweet.all
+      erb :'/tweets/index'
+    end
+  end
+
+  get '/tweets/new' do
+    if Helpers.is_logged_in?(session)
+      erb :'/tweets/new'
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content].empty?
+      redirect to '/tweets/new'
+    else
+      @tweet = Tweet.create(content: params[:content], user_id: session[:user_id])
+      # if params[:phone_number] != ""
+      #   @client = Twilio::REST::Client.new(ACCOUNT_SID, AUTH_TOKEN)
+      #   number = "#{params[:phone_number]}"
+      #   @client.messages.create(
+      #     from: 'Number goes here',
+      #     to: number,
+      #     body: params[:content] + "\n - sent by #{@tweet.user.username} from Fwitter"
+      #     )
+      # end
+      redirect to "/tweets/#{@tweet.id}"
+    end
+  end
+
+  get '/tweets/:id' do #tweet show page
+    if !Helpers.is_logged_in?(session)
+      redirect to '/login'
+    else
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'/tweets/show'
+    end
+  end
+
+  get '/tweets/:id/edit' do #tweet edit page
+    if !Helpers.is_logged_in?(session)
+      redirect to '/login'
+    else
+    @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == session[:user_id]
+        erb :'/tweets/edit'
+      else
+        redirect to '/tweets'
+      end
+    end
+  end
+
+  patch '/tweets/:id' do
+    @tweet = Tweet.find_by_id(params[:id])
+
+    if !params[:content].empty?
+      @tweet.update(content: params[:content])
```

Comment:
then, for example, this would be a good place to use AR's has_many method, to associate the user with the tweet on creation?  then verify that @tweet.user = current_user?